### PR TITLE
Refactor: use PayPal Client for api requests

### PR DESCRIPTION
**Pull request by ravinderk**
Resolves https://github.com/impress-org/givewp/pull/6785#issuecomment-1529365486

## Description

I found that we are making a few direct rest API requests to PayPal, which a few times cause an issue because of the expired merchant access token. I update these requests to use PayPal Client, which handles expired merchant access tokens internally.

## Affects

This pull request affects the following feature:
- PayPal merchant onboarding
- Refresh action token
- Delete PayPal account connection

## Testing Instructions

- [ ] Admin should able to onboard successfully with live and sandbox accounts.
- [ ] Admin should able to disconnect live and sandbox account connection.
- [ ] Admin should be able to activate and deactivate recurring add-ons.
- [ ] Access token should refresh correctly.
- [ ] Subscription should renew correctly.

## Pre-review Checklist

-   [x] Acceptance criteria satisfied and marked in related issue
-   [ ] Relevant `@unreleased` tags included in DocBlocks
-   [ ] Includes unit tests
-   [ ] Reviewed by the designer (if follows a design)
-   [x] [Self Review](https://give.gitbook.io/development-manual/devops/github/code-reviews#self-review) of code and UX completed

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1203760890899570
  - https://app.asana.com/0/0/1205157507197551
  - https://app.asana.com/0/0/1204987672426327
  - https://app.asana.com/0/0/1204621958568566

### CODE REVIEW

**Review by JasonTheAdams — CHANGES_REQUESTED** [review]
Looks good! Just a couple very minor changes!

File: src/PaymentGateways/PayPalCommerce/Repositories/PayPalAuth.php [hunk 1]
```diff
@@ -99,27 +95,36 @@ public function getTokenFromClientCredentials($client_id, $client_secret): array
...
+                [
+                    'category' => 'Payment Gateway',
+                    'source' => 'Paypal Commerce',
+                    'response' => $response,
                 ]
-            )
-        );
+            );
+
+            throw new Exception('PayPal Commerce: Error retrieving access token with authorization code');
```
[the remark this answers is not in the document]
Comment (ravinderk) [reply src/PaymentGateways/PayPalCommerce/Repositories/PayPalAuth.php]: @JasonTheAdams I used `RuntimeException`.

I also used `UnexpectedValueException` when getting invalid access or a seller token.

File: src/PaymentGateways/PayPalCommerce/PayPalClient.php [hunk 2]
```diff
@@ -50,6 +50,7 @@ public function getHttpClient(): PayPalHttpClient
     /**
      * Get api url.
      *
+     * @deprecated x.x.x
```
[the remark this answers is not in the document]
Comment (ravinderk) [reply src/PaymentGateways/PayPalCommerce/PayPalClient.php]: fixed: https://github.com/impress-org/givewp/pull/6819/commits/ab536817894274a9b864631de8884d3aa291ceae

**Review by JasonTheAdams — APPROVED** [review]
Great! Please shoot this over to QA!

**Review by rickalday — APPROVED** [review]
Passed manual tests

**Pull request merged**